packages/core/test/new-api/helpers.ts

```diff
@@ -399,9 +400,39 @@ export class MockChainDispatcher implements ChainDispatcher {
     return response as any;
   }
 
+  public async getTransaction(
+    txHash: string
+  ): Promise<ethers.providers.TransactionResponse | null | undefined> {
+    const [from, nonce] = txHash.split("--");
```

Comment:
What's this? I don't get it.

Reply:
The `sendTx` above is generating fake hashes based on:
```
const hash = `${from}--${tx.nonce?.toString() ?? "no-nonce"}`
```

This is pulling the from/nonce from this as a way of smuggly information within the mock.